metis-enrichment/metis-enrichment-service/src/test/java/eu/europeana/enrichment/service/EnrichmentServiceTest.java

```diff
@@ -0,0 +1,214 @@
+  }
+
+  @Test
+  void enrichByEquivalenceValues_throwsException() {
+    final ReferenceValue referenceValue = new ReferenceValue("invalidUrl",
+        Set.of(EntityType.AGENT, EntityType.TIMESPAN));
+    final BadRequestException badRequestException = assertThrows(BadRequestException.class,
+        () -> enrichmentService.enrichByEquivalenceValues(referenceValue));
+    assertEquals(MalformedURLException.class, badRequestException.getCause().getClass());
+  }
+
+  @Test
+  void enrichById() throws MalformedURLException {
+    final String entityAbout = enrichmentObjectUtils.agentTerm1.getEnrichmentEntity().getAbout();
+    final ReferenceTerm referenceTerm = new ReferenceTermImpl(new URL(entityAbout),
+        new HashSet<>());
+    final EnrichmentBase agentEnrichmentBase = Converter.convert(enrichmentObjectUtils.agentTerm1);
+    assertNotNull(agentEnrichmentBase);
+    final Map<ReferenceTerm, EnrichmentBase> referenceTermListHashMap = new HashMap<>();
+    referenceTermListHashMap.put(referenceTerm, agentEnrichmentBase);
+    when(persistentEntityResolver.resolveById(Set.of(referenceTerm)))
+        .thenReturn(referenceTermListHashMap);
+    final EnrichmentBase enrichmentBase = enrichmentService.enrichById(entityAbout);
+
+    assertNotNull(enrichmentBase);
+    assertEquals(agentEnrichmentBase.getAbout(), enrichmentBase.getAbout());
+    verify(persistentEntityResolver, times(1)).resolveById(Set.of(referenceTerm));
+  }
+
+  @Test
+  void enrichById_throwsException() {
+    final String entityAbout = "InvalidUrl";
+    final BadRequestException badRequestException = assertThrows(BadRequestException.class,
+        () -> enrichmentService.enrichById(entityAbout));
+    assertEquals(MalformedURLException.class, badRequestException.getCause().getClass());
+  }
+
+  @Test
+  void saveOrganization() {
+    final OrganizationEnrichmentEntity organizationEnrichmentEntity = (OrganizationEnrichmentEntity) enrichmentObjectUtils.organizationTerm1
+        .getEnrichmentEntity();
+    final Date created = new Date();
+    final Date updated = new Date();
+    when(persistentEntityResolver.saveOrganization(organizationEnrichmentEntity, created, updated))
+        .thenReturn(organizationEnrichmentEntity);
+    final OrganizationEnrichmentEntity organizationEnrichmentEntityResult = enrichmentService
+        .saveOrganization(organizationEnrichmentEntity, created, updated);
+
+    assertNotNull(organizationEnrichmentEntityResult);
+    assertEquals(organizationEnrichmentEntity.getAbout(),
+        organizationEnrichmentEntityResult.getAbout());
+  }
+
+  @Test
+  void findExistingOrganizations() {
+    final List<String> toSearch = List
+        .of(enrichmentObjectUtils.organizationTerm1.getEnrichmentEntity().getAbout(),
+            enrichmentObjectUtils.customOrganizationTerm.getEnrichmentEntity().getAbout());
+    when(persistentEntityResolver.findExistingOrganizations(toSearch)).thenReturn(
```

Comment:
No it's just the  nature of the test, I'm search 2 but I get 1 because only 1 was available..
But I see now that this might be confusing so I've updated it to return and check for both